pages/posts/get-started-with-json-schema-in-node-js.md

```diff
@@ -0,0 +1,179 @@
+
+```json
+{ "type": "string" }
+```
+
+To validate data against a schema, we need to use a validator library which implements the JSON Schema specification. The [Ajv (Another JSON Schema Validator)](https://ajv.js.org/) library is the most popular JSON Schema validator for client and server side JavaScript, downloaded over [50 million times every week](https://www.npmtrends.com/ajv) from npm.
+
+Let's get hands-on and see what validating data with JSON Schema and Ajv looks like.
+
+> For a more detailed introduction to JSON Schema, check out the blog post [JSON Schema in 5 minutes](https://json-schema.org/blog/posts/json-schema-in-5-minutes).
+
+## Creating a schema and validating data
+
+First we need to run the command `npm install ajv` to install the Ajv library.
+
+Then we can jump straight into defining a JSON schema. It describes the structure and types we expect:
+
+```javascript
+const iceCreamSchema = {
+  type: "object",
+  properties: {
+    flavour: { type: "string" },
+    price: { type: "number" },
+    stock: { type: "number" },
+  },
+  required: ["flavour", "price", "stock"],
+};
+```
+
+Now we'll define the data which we want to validate:
+
+```javascript
+const iceCreamData = {
+  flavour: "Pistachio",
+  price: 1.99,
+  stock: null,
+};
+```
+
+Then we'll import the Ajv library and create a new Ajv instance:
+
+```javascript
+import Ajv from "ajv";
+
+const ajv = new Ajv();
+```
+
+And use it to validate the data against our schema:
+
+```javascript
+const isDataValid = ajv.validate(iceCreamSchema, iceCreamData);
+```
+
+Lastly, we'll add some code to handle the validation results:
+
+```javascript
+if (isDataValid) {
+  console.log("The ice cream data is valid! 🍨");
+} else {
+  console.error("The ice cream data is invalid:", ajv.errors);
+}
+```
+
+When we put this code together and run it, we get the following output:
+
+```
+The ice cream data is invalid: [
+  {
+    instancePath: '/stock',
+    schemaPath: '#/properties/stock/type',
+    keyword: 'type',
+    params: { type: 'number' },
+    message: 'must be number'
+  }
+]
+```
+
+Pretty slick, huh?
+
+Alright, we've created a schema which describes the structure and types we expect, validated data against it, and handled any validation errors. So far so good!
+
+As one final step, we'll look at how we can potentially improve the way we create our JSON schema.
+
+## Schema generation
+
+JSON schemas are an expressive way of defining validation rules, but writing schemas "by hand" can sometimes get a bit much. There's a handy library called [fluent-json-schema](https://npm.im/fluent-json-schema) which can help us generate our JSON schemas. Let's give it a try.
+
```

Comment:
```suggestion

Note that while using a tool to define validaiton rules to be converted into a JSON Schema gives you control, other methods of generating JSON Schema, for example from your instance data, often leaves you with a partially complete or sometimes incorrect JSON Schema. Other approaches might be better called "scaffolding" rather than "generating", as you'll have the basics, but still have work to do before it's complete and useable.
```